cirq/google/engine/engine.py

```diff
@@ -213,7 +207,7 @@ def __init__(self,
     def run(
             self,
             *,  # Force keyword args.
-            program: Program,
+            program: Dict,
```

Comment:
I think we want to leave the ability to run directly from engine object which does create program, create job, return result.  But we want to add a run onto the program object so that it can be called repeatedly.  That way we don't break backwards compatibility.

Reply:
Fixed, except that the program_id is removed from JobConfig, as discussed offline.